### ISSUE TITLE
Warning not provided for obsolete method ReadOnlySpan.Equals

### ISSUE THREAD

**Issue by sweemer** [dotnet/roslyn#76632]
**Version Used**: 

```console
$ dotnet --info
.NET SDK:
 Version:           9.0.101
 Commit:            eedb237549
 Workload version:  9.0.100-manifests.4a280210
 MSBuild version:   17.12.12+1cce77968

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\9.0.101\
```

**Steps to Reproduce**:

1. `git clone https://github.com/sweemer/ReadOnlySpanEquals.git` 
2. `cd ReadOnlySpanEquals && dotnet run`
3. `Unhandled exception. System.NotSupportedException: Equals() on Span and ReadOnlySpan is not supported. Use operator== instead.`

**Diagnostic Id**: CS0618

**Expected Behavior**: Warning shown when building project.

**Actual Behavior**: No warning shown, only runtime exception thrown.

Note that IntelliSense in Visual Studio Code correctly flags this method as deprecated, but no warning is shown.

![Image](https://github.com/user-attachments/assets/261785d5-58f3-4f84-8d14-52f42fce7e53)

**Comment by AlekseyTs** [comment]
It looks like the behavior is "By Design", obsolete state of a method is determined based on the least overridden declaration (Object.Equals in this case). There is a comment in ```Binder.ReportDiagnosticsIfObsolete```:
```
            // There are two reasons to walk up to the least-overridden member:
            //   1) That's the method to which we will actually emit a call.
            //   2) We don't know what virtual dispatch will do at runtime so an
            //      overriding member is basically a shot in the dark.  Better to
            //      just be consistent and always use the least-overridden member.
```

Compiler also reports a warning when an Obsolete attribute is placed on an override:
```
warning CS0809: Obsolete member 'S1.Equals(object)' overrides non-obsolete member 'object.Equals(object?)'
```

Also, apparently the ReadOnlySpan.Equals implementation violates the following rule:
```
error CA1065: Equals creates an exception of type NotSupportedException. Exceptions should not be raised in this type of method. If this exception instance might be raised, change this method's logic so it no longer raises an exception. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1065)
```

I am not sure why framework decided to ignore all the "red" flags and applied the Obsolete attribute despite the fact that its presence does not make a difference. Since a span cannot be be boxed, it feels like always returning `false` would be correct implementation.

**Comment by sweemer** [comment]
@AlekseyTs Thanks for the thorough explanation. Can this issue be forwarded to the team that maintains the framework to see if there is any room for improvement?

**Comment by AlekseyTs** [comment]
Transferring to runtime
